ngraph/core/reference/include/ngraph/runtime/reference/roll.hpp

```diff
@@ -0,0 +1,61 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/coordinate_transform.hpp"
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename T>
+            void roll(const T* arg,
+                      const int64_t* shift,
+                      const int64_t* axes,
+                      T* out,
+                      const Shape& arg_shape,
+                      const Shape& shift_shape,
+                      const Shape& axes_shape)
+            {
+                std::vector<int64_t> axes_vector = std::vector<int64_t>(axes, axes + axes_shape[0]);
+                for (auto& axis : axes_vector)
+                {
+                    if (axis < 0)
+                        axis += arg_shape.size();
+                }
+
+                CoordinateTransform arg_transform{arg_shape};
```

Comment:
Our current approach is to avoid use of `CoordinateTransform` and implement manual data manipulation. @ilyachur , please correct me if I am wrong.

Reply:
Yes you are right